### PR TITLE
[For 2.7.1] fix(backend): fix default value for parameters and others config file fix

### PR DESCRIPTION
**Pull request by inkhey**
related to #2920 

## Checkpoints

The goal of this section is to help code reviewer and merge request author to do required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

**For code reviewer**

- [x] Code is clear enough
- [x] If there are FIXME in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODO, NOTE or HACK in code, date and developer initials are present
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation

**For developper**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] Original issue have been updated with a short resume of implemented solution

**For quality team**

- [x] Tested by quality team

### CODE REVIEW

File: backend/development.ini.sample [hunk 1]
```diff
@@ -513,17 +539,22 @@ webdav.listen = localhost:3030
 # This feature require application "agenda" in "app.enabled"
 
 ### Radicale Proxy config ##
-## path to Tracim radicale server, usually at localhost:port
-caldav.radicale_proxy.base_url = http://localhost:5232
+## path to Tracim radicale server, usually at localhost:por
```
Comment (inkhey) [inline backend/development.ini.sample:542]:
```suggestion
## path to Tracim radicale server, usually at localhost:port
```

File: backend/tracim_backend/config.py [hunk 2]
```diff
@@ -364,7 +364,7 @@ def _load_global_config(self) -> None:
         self.API__KEY = self.get_raw_config("api.key", "", secret=True)
         self.SESSION__REISSUE_TIME = int(self.get_raw_config("session.reissue_time", "120"))
         default_session_data_dir = self.here_macro_replace("%(here)s/sessions_data")
-        default_session_lock_dir = self.here_macro_replace("%(here)s/sessions_data")
+        default_session_lock_dir = self.here_macro_replace("%(here)s/sessions_lock")
```
Comment (PhilippeAccorsi) [inline backend/tracim_backend/config.py:367]: This change can be break some existing connexion when upgrade of tracim is done? If yes, what user need to do ?

**Comment by PhilippeAccorsi** [comment]
Test need to be done correctly with docker image. Test elasticsearch and collaboraonline to make its working correctly.

Check also if email working and if list off enabled app works good.
To test correctly, its necessary to start docker multiple time with no old value in config file or docker volume. Need also test when starting docker with existing parameter in config file (ex: activate collabora, test if its working. deactivate collabora, test, activate again).

We need to be sure all working correctly for tracim 2.7.1

File: backend/tracim_backend/config.py (hunk 2 above)
Comment (inkhey) [reply backend/tracim_backend/config.py:367]: 1. It's very unlikely that some user use this config has this is the code default value but defaut config file have a value setted (basic_setup) which is different (and correct)
2. It's not a big deal if existing session are dropped.

File: backend/development.ini.sample [hunk 3]
```diff
@@ -38,6 +48,10 @@ basic_setup.session_secret = change_this_value_please!
 # CORE
 ####
 
+# hapic debug mode: return python traceback of error
+; debug = False
```
Comment (buxx) [inline backend/development.ini.sample:52]: This parameter only configure hapic debug mode ? If yes, this is a shadow config name no ?

File: backend/development.ini.sample [hunk 4]
```diff
@@ -153,22 +170,19 @@ preview_cache_dir = %(basic_setup.preview_cache_dir)s
...
-advanced_setup.session_lock_dir = %(basic_setup.sessions_data_root_dir)s/sessions_lock
 
 ### pyramid_beaker parameters ###
 session.type = file
-session.data_dir = %(advanced_setup.session_data_dir)s
-session.lock_dir = %(advanced_setup.session_lock_dir)s
+; session.data_dir = %(here)s/sessions_data
+session.data_dir = %(basic_setup.sessions_data_root_dir)s/sessions_data
+; session.lock_dir = %(here)s/sessions_data
+session.lock_dir = %(basic_setup.sessions_data_root_dir)s/sessions_lock
```
Comment (buxx) [inline backend/development.ini.sample:185]: Why do not comment them and let default value ?

Comment (inkhey) [reply backend/development.ini.sample:185]: That the decision we have choosen to clarify that default value is different
see solution 1 of https://github.com/tracim/tracim/issues/2925

File: backend/development.ini.sample (hunk 3 above)
Comment (inkhey) [reply backend/development.ini.sample:52]: "shadow" ?

File: backend/development.ini.sample [hunk 5]
```diff
@@ -190,9 +204,10 @@ session.cookie_on_exception = True
 
 ## Api key auth is disabled by default,
 ## you should set a not-null value to api.key to active it:
+; api.key =
 api.key = %(basic_setup.api_key)s
 ## base url of the backend API, default to same as website.base_url
-; api.base_url = http://localhost:6543
+; api.base_url =
```
Comment (buxx) [inline backend/development.ini.sample:210]: There is no default value here ?

File: backend/development.ini.sample (hunk 4 above)
Comment (inkhey) [reply backend/development.ini.sample:185]: solution 2 is possible but more complicated. We will check this another time.

File: backend/development.ini.sample (hunk 5 above)
Comment (inkhey) [reply backend/development.ini.sample:210]: generated default value are showed as empty to don't make user think that default value is fixed one.

File: backend/development.ini.sample [hunk 6]
```diff
@@ -243,37 +257,39 @@ api.key = %(basic_setup.api_key)s
...
+## if you want to use Tracim API without frontend, change this parameter
 ; frontend.serve = True
-# You can set dist folder of Tracim frontend. by default, system
+# You can set dist folder of Tracim frontend. By default, system
 # will try to get it automatically according to Tracim repository
 # organisation.
-; frontend.dist_folder_path = /home/user/tracim/frontend/dist
+; frontend.dist_folder_path =
+# example:
+# frontend.dist_folder_path = /home/user/tracim/frontend/dist
```
Comment (buxx) [inline backend/development.ini.sample:289]: Whe backend does not use an `(here)` mechanism ?

Comment (inkhey) [reply backend/development.ini.sample:289]: That's and example.

File: backend/development.ini.sample [hunk 7]
```diff
@@ -2,8 +2,18 @@
 # Tracim Config File #
 ######################
 
+## How work this file
+#
+# Tracim use all parameters not commented here (environnement variable is used first if define).
+# Lines who start by `;` is used to comment parameters and show default value available in Tracim code.
+# Lines who start by `#` is used for comment and to show parameters examples.
```
Comment (grignards) [inline backend/development.ini.sample:9]: Petite retouche des explications :

## Configuration
# Parameters can be defined via environment variables or in the config file.
# Environment parameters are used in priority if a variable is defined in both.

## File Syntax
# Comments start by `;` or `#`.
# In the sample config file, `;` commented line are used to indicate the default values used by Tracim;
# `#` commented lines are used for comments/examples.
# Boolean values can be `True` or `False`
# All parameters starting with `basic_setup` can be used as variables in other parameters. They are all mandatory.

Comment (inkhey) [reply backend/development.ini.sample:9]: @grignards I do agree for almost all but basic_setup parameter are not really mandatory, it's just shortcuts to allow setup basic setup tracim parameters without needing to read the all configuration file. So we should probably remove the "mandatory" part.

Comment (grignards) [reply backend/development.ini.sample:9]: OK, I rewrote this part: `For simple needs they are all you need to customize`.

Comment (inkhey) [reply backend/development.ini.sample:9]: +1:

File: backend/development.ini.sample [hunk 8]
```diff
@@ -364,27 +380,26 @@ website.base_url = %(basic_setup.website_base_url)s
...
-## valid variable is 'content_id' for 'reply_to' header:
-; email.notification.reply_to.email =
-## valid variable is 'content_id' for 'references' header:
-; email.notification.references.email =
-
 # example of recommanded pattern (for 'from' header):
 # email.notification.from.email = noreply+{user_id}@domainname.tld
 
+## valid variable is 'content_id' for 'reply_to' header:
+; email.notification.reply_to.email =
```
Comment (buxx) [inline backend/development.ini.sample:393]: Can you add an example ?

File: backend/development.ini.sample [hunk 9]
```diff
@@ -364,27 +380,26 @@ website.base_url = %(basic_setup.website_base_url)s
...
 # tagging reply_to header of email with content_id is needed for working
 # email_reply feature
 # example of recommanded pattern (for 'reply_to' header):
 # email.notification.reply_to.email = reply+{content_id}@domainname.tld
 # email.notification.reply_to.email = {content_id}@reply.domainname.tld
 # email.notification.reply_to.email = reply+cid{content_id}@domainname.tld
 
+## valid variable is 'content_id' for 'references' header:
+; email.notification.references.email =
```
Comment (buxx) [inline backend/development.ini.sample:402]: Can you add an example here ?

File: backend/development.ini.sample (hunk 3 above)
Comment (buxx) [reply backend/development.ini.sample:52]: > "shadow" ?

https://en.wikipedia.org/wiki/Variable_shadowing#Python

I mean: `debug` name look like a config to enable debug on whole Tracim. But, like indicated by comment, it enable only debug for hapic.

File: backend/development.ini.sample (hunk 9 above)
Comment (inkhey) [reply backend/development.ini.sample:402]: There is example just after.

File: backend/development.ini.sample (hunk 8 above)
Comment (inkhey) [reply backend/development.ini.sample:393]: There are examples just after

File: backend/development.ini.sample (hunk 3 above)
Comment (inkhey) [reply backend/development.ini.sample:52]: En fait y'a pas de variable shadowing car on à pas d'autre debug dans tracim.

Comment (inkhey) [reply backend/development.ini.sample:52]: le reste du debug c'est la configuration des logger c'est autre chose.

**Comment by inkhey** [comment]
> Test need to be done correctly with docker image. Test elasticsearch and collaboraonline to make its working correctly.
> 
> Check also if email working and if list off enabled app works good.
> To test correctly, its necessary to start docker multiple time with no old value in config file or docker volume. Need also test when starting docker with existing parameter in config file (ex: activate collabora, test if its working. deactivate collabora, test, activate again).
> 
> We need to be sure all working correctly for tracim 2.7.1

Tested with multiple configuration with defaut base docker:
- with elasticsearch
- with email activated
- with collabora.
Tested also restart with these feature disable using env var, work as expected.

File: backend/development.ini.sample (hunk 7 above)
Comment (grignards) [reply backend/development.ini.sample:9]: Is that ok with you if I push a commit with those changes ?

**Comment by lebouquetin** [comment]
@inkhey this Merge request must be tested and validated and merged on wednesday, april, the 29th. You must do all what you can for this (even ask @buxx and @grignards for talk if required). @PhilippeAccorsi is back on thurday, I expect he will be able to tag znd deliver new release in the early morning

File: backend/development.ini.sample (hunk 7 above)
Comment (inkhey) [reply backend/development.ini.sample]: ok, thanks.

**Comment by inkhey** [comment]
> > Test need to be done correctly with docker image. Test elasticsearch and collaboraonline to make its working correctly.
> > Check also if email working and if list off enabled app works good.
> > To test correctly, its necessary to start docker multiple time with no old value in config file or docker volume. Need also test when starting docker with existing parameter in config file (ex: activate collabora, test if its working. deactivate collabora, test, activate again).
> > We need to be sure all working correctly for tracim 2.7.1
> 
> Tested with multiple configuration with defaut base docker:
> 
>     * with elasticsearch
> 
>     * with email activated
> 
>     * with collabora.
>       Tested also restart with these feature disable using env var, work as expected.
> 
> 
> Tested with existing base too, do run correctly.

Tested with old config file and some tweak to make thing work. Does work as expected.
Tried multiple config change too.

Not fully tested point now are reply by email and collabora as both need specific network configuration ( email fetcher should be able to call tracim itself with same url as website.base_url and elasticsearch need server to be both avaiable at same url from outside and inside of tracim).

**Comment by inkhey** [comment]
> Not fully tested point now are reply by email and collabora as both need specific network configuration ( email fetcher should be able to call tracim itself with same url as website.base_url and elasticsearch need server to be both avaiable at same url from outside and inside of tracim).

This is now tested.

**Comment by inkhey** [comment]
Just added a fix for non-working email translation in both docker and standard tracim.

**Pull request merged**